src/common/IsolatedWorld.ts

```diff
@@ -168,12 +168,15 @@ export class IsolatedWorld {
     this.#contextPromise = createDeferredPromise();
   }
 
-  setContext(context: ExecutionContext): void {
+  async setContext(context: ExecutionContext): Promise<void> {
     assert(
       this.#contextPromise,
       `ExecutionContext ${context._contextId} has already been set.`
     );
     this.#ctxBindings.clear();
+    if (this.#injected) {
+      await context.evaluate(injectedSource);
```

Comment:
I think that might create a race as the calling code might expect the context to be available immediately after setContext. What about we keep it sync and just schedule the injection?

Reply:
Hm, I think it might be that some `#onBindingCalled` messages are missing because the script is being evaluates. Let's see if tests catch anything.

Reply:
I'll do this in a followup PR.

Reply:
Obsolete.